src/main/java/zd/zero/waifu/motivator/plugin/promotion/MemePromotionDialog.kt

```diff
@@ -3,7 +3,7 @@ package zd.zero.waifu.motivator.plugin.promotion
 import com.intellij.ide.BrowserUtil
 import com.intellij.openapi.extensions.PluginId
 import com.intellij.openapi.ui.DialogWrapper
-import com.intellij.openapi.updateSettings.impl.pluginsAdvertisement.PluginsAdvertiser
+import com.intellij.openapi.updateSettings.impl.pluginsAdvertisement.installAndEnable
```

Comment:
I'm not 100% sure this is backwards compatible with builds lower than 2021.2. The lowest build supported right now is 2020.1.2 ? (whatever 202. is)

I haven't tested it out though, I might try sometime this weekend though, and report back.

Reply:
Ah Right! Imma test this build

Reply:
[`202.*` is `2020.2` platform version](https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html#intellij-platform-based-products-of-recent-ide-versions). Passes in the plugin verifier with `2020.2.4` and downloaded the older IDE version (`idePath=/home/zaerald/Downloads/idea-IC-202.6948.69`:`2020.2.1`)

All these builds and versions makes me dizzy 😵 lol

Reply:
Can confirm, I ran it on

```
IntelliJ IDEA 2020.2.1 (Community Edition)
Build #IC-202.6948.69, built on August 24, 2020
Runtime version: 11.0.8+10-b944.31 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.4.0-80-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 512M
Cores: 12
Non-Bundled Plugins: io.unthrottled.amii, zd.zero.waifu-motivator-plugin
Current Desktop: ubuntu:GNOME
```
Was able to install the plugin through the dialog. Good to know! The verifier said it was fine, but I was paranoid it wouldn't work (the file was converted from java to kotlin from 2021.1 to 2021.2). Now I know!

Side note, I did observe this on dynamically installing WMP, dunno if you saw this as well. It prevented the initial plugin install prompt, but did show up after I reloaded the project.

<details>
<summary>Error on install</summary>

```
Error executing task zd.zero.waifu.motivator.plugin.listeners.PluginInstallListener$$Lambda$3103/0x0000000101ed4440@3e30de9c

java.lang.NoSuchMethodError: 'com.intellij.notification.Notification com.intellij.notification.NotificationGroup.createNotification(java.lang.String, java.lang.String, com.intellij.notification.NotificationType)'
	at zd.zero.waifu.motivator.plugin.onboarding.UpdateNotification.display(UpdateNotification.kt:48)
	at zd.zero.waifu.motivator.plugin.onboarding.UserOnboarding.attemptToPerformNewUpdateActions$lambda-0(UserOnboarding.kt:18)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at zd.zero.waifu.motivator.plugin.onboarding.UserOnboarding.attemptToPerformNewUpdateActions(UserOnboarding.kt:16)
	at zd.zero.waifu.motivator.plugin.listeners.PluginInstallListener.pluginLoaded$lambda-1$lambda-0(PluginInstallListener.kt:30)
	at com.intellij.openapi.project.DumbServiceImpl.doRun(DumbServiceImpl.java:348)
	at com.intellij.openapi.project.DumbServiceImpl.updateFinished(DumbServiceImpl.java:340)
	at com.intellij.openapi.project.TrackedEdtActivityService$TrackedEdtActivity.run(TrackedEdtActivityService.java:100)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:201)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:802)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:322)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:84)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:132)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:47)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:188)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:971)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:841)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:744)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:451)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:802)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:499)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```
</details>

Reply:
Resolved in PR #356 
I've reverted the `platformVersion` to `2020.2.4`. Even though it could be installed properly to `2021.2` build, it'll fail in runtime.